lib/src/collect_lcov.dart

```diff
@@ -102,12 +102,43 @@ class LcovCollector {
   /// Generates and returns a coverage json file
   Future<CoverageResult<List<File>>> _getCoverageJson(
       String testFile, Directory coverageDir) async {
-    var args = [
-      "--coverage_dir=${coverageDir.path}",
-      "--package-root=${packageRoot}",
+
+    String observatoryPort = "8181";
+
+    var dartArgs = [
+      "--pause-isolates-on-exit",
+      "--enable-vm-service=${observatoryPort}",
       testFile
     ];
-    var result = await processSystem.runProcess(Platform.executable, args);
+
+    processSystem.runProcess(Platform.executable, dartArgs).then((result){
+      if (result.exitCode < 0) {
+        stderr.writeln('stdout:');
+        stderr.writeln(result.stdout);
+        stderr.writeln('stderr:');
+        stderr.writeln(result.stderr);
+        throw new ProcessException(Platform.executable, dartArgs,
+            'There was a critical error. Exit code: ${result.exitCode}',
+            result.exitCode);
+      }
+    });
+
+    var cvArgs = [
+      "global",
```

Comment:
It was the easiest way to get to a result that `dart_coveralls` can consume without refactoring, but I'm halfway through of getting it to work via direct invocation of `coverage` API methods.